packages/next/server/dev/next-dev-server.ts

```diff
@@ -305,7 +305,9 @@ export default class DevServer extends Server {
         ignored: (pathname: string) => {
           return (
             !files.some((file) => file.startsWith(pathname)) &&
-            !directories.some((dir) => pathname.startsWith(dir))
+            !directories.some(
+              (dir) => pathname.startsWith(dir) || dir.startsWith(pathname)
+            )
```

Comment:
it was that? thanks much @ijjk I never though that the problem was that, I was searching for something in webpack config file all this time but without success

Reply:
Yeah seems we were ignoring the `src` directory incorrectly for this case here

Reply:
But I dont understand, why it worked with the empty pages folder inside the src folder then, it is related?